src/main/resources/fetch-artifact.template.html

```diff
@@ -2,4 +2,9 @@
     <label ng-class="{'is-invalid-label': GOINPUTNAME[EnvironmentVariablePrefix].$error.server}">Environment Variable Prefix:</label>
     <input ng-class="{'is-invalid-input': GOINPUTNAME[EnvironmentVariablePrefix].$error.server}" type="text" ng-model="EnvironmentVariablePrefix" placeholder="PREFIX"/>
     <span class="form_error form-error" ng-class="{'is-visible': GOINPUTNAME[EnvironmentVariablePrefix].$error.server}" ng-show="GOINPUTNAME[EnvironmentVariablePrefix].$error.server">{{GOINPUTNAME[EnvironmentVariablePrefix].$error.server}}</span>
-</div>
\ No newline at end of file
+</div>
+<div class="form_item_block">
+    <label ng-class="{'is-invalid-label': GOINPUTNAME[SkipImagePulling].$error.server}">Skip Image Pulling:</label>
+    <input ng-class="{'is-invalid-input': GOINPUTNAME[SkipImagePulling].$error.server}" type="checkbox" ng-model="SkipImagePulling"/>
```

Comment:
Hi @varshavaradarajan, I have moved the label behind the checkbox, but have issues working with the checkbox values. It is still sticking to `on` even though the ng-*-value attributes were added. Also the checkbox is not filled out when the config was set, I feel like I'm missing some part in the frontend. Can you give me a hint what might be the issue here?

Reply:
@alexfrancavilla - I don't know why either. Sorry. Maybe @bdpiparva can help.

Reply:
@alexfrancavilla This seems to do the trick:

```html
<input id="SkipImagePullingInput" ng-class="{'is-invalid-input': GOINPUTNAME[SkipImagePulling].$error.server}" value="{{SkipImagePulling}}" type="checkbox" ng-model="SkipImagePulling" ng-true-value="true" ng-false-value="false"/>
```

I added:

```
value="{{SkipImagePulling}}"
```

along with: ```ng-true-value="true" ng-false-value="false"```.

@bdpiparva would know for sure, but what seems to be happening is the ```element.value``` of that DOM element continued to be set to `on`, even though the Angular model (`ng-model`) was correct. So, when it was serialized (by Rails), that was the value is sent along. I just forced the value to be set to `true` and when serialized, it sends `true` or `false`.

If we really wanted to make it do the right thing, without this, we would need to do what is mentioned [here](https://apidock.com/rails/ActionView/Helpers/FormHelper/check_box) and have another hidden value, etc. I don't think it's worth it.

In this particular case, if you un-check the checkbox, no value gets sent for that checkbox, to the server, and it ends up removing it from the config, instead of marking it as `false`. It works, because `null` is also the same as `false` in this case.

If you think it makes sense, please change it, @alexfrancavilla.

Reply:
@arvindsv - had tried this. If you check the config xml, the value is stored as `'true'` and not `true`. So, if someone happens to modify it via the API to `true` and then simply opens the modal, the value is unchecked and saving it means that, the value ends up being null and defaults to false. I don't know if that's alright. Or if the whole pipeline is created via a config-repo or API, and `true` is passed instead of `'true'`, it should be handled by the plugin, right?  

 I had also tried `value="true"`. But even though it gets stored as `true` in the xml, the checkbox is always unchecked. 

I don't know of many scm materials and task plugins that have the plugin view in rails with a checkbox. I checked slack task plugin and there, `ng-true-value` seemed to work - https://github.com/Vincit/gocd-slack-task/blob/master/src/main/resources/views/task.template.html#L62. So, I'm not sure if something is wrong with the way the plugin view is rendered in the fetch task modal. 

Anyway, maybe, checking for both `on` and `true` could solve the problem instead of introducing `value`?

Reply:
@varshavaradarajan I don't see that. I thought it was because I had played around with change the type of `skipImagePulling` to `boolean` from `String`, but it still has the same value in the XML, with or without that change:

```
<property>
  <key>SkipImagePulling</key>
  <value>true</value>
</property>
```

Reply:
@arvindsv is right  adding `ng-true-value` and `ng-false-value` with value will work fine.

```html
<div class="form_item_block" ng-init="{SkipImagePulling = SkipImagePulling || false}">
    <input id="SkipImagePullingInput" ng-class="{'is-invalid-input': GOINPUTNAME[SkipImagePulling].$error.server}" type="checkbox"
		   value="{{SkipImagePulling}}" ng-model="SkipImagePulling" ng-true-value="true" ng-false-value="false"/>
    <label for="SkipImagePullingInput" ng-class="{'is-invalid-label': GOINPUTNAME[SkipImagePulling].$error.server}">Skip Image Pulling</label>
    <span class="form_error form-error" ng-class="{'is-visible': GOINPUTNAME[SkipImagePulling].$error.server}" ng-show="GOINPUTNAME[SkipImagePulling].$error.server">{{GOINPUTNAME[SkipImagePulling].$error.server}}</span>
</div>
```

![check-box-issue](https://user-images.githubusercontent.com/7871209/50466669-f3cc4000-09c4-11e9-8714-01dcf52b31d0.gif)


PS: `ng-init="{SkipImagePulling = SkipImagePulling || false}"` - is used to set default value(not needed in this scenario)

Reply:
@arvindsv - you were right. I was getting the quotes because ng-true-value had `''` in them. This works. Sorry about that.

Reply:
Sorry for the delay, I've cought something flu-like and was ill for a longer time than I hoped for.

@arvindsv @varshavaradarajan  @bdpiparva Thank you very much for your input! I've added the `value`, `ng-true-value` and `ng-false-value` to the checkbox and confirm that this combination works how we wanted it.

Furthermore I've changed the type of the config flag to `boolean` in the `FetchArtifactConfig` class. I was not so super sure about how things work in detail there - I'm thinking that the `@FieldMetadata` should remain on the default String type since that mechanism only validates the configuration xml which is always a string. So I changed the field and getter to parse the value as a boolean. Is this approach correct or am I missing something? (see cece8dd)

Reply:
@alexfrancavilla I think the change to `boolean` is correct. I have no issues with it.

@varshavaradarajan ?